Alternative installation server doc (#5844)
Scille/parsec-cloud: docs/adminguide/hosting/index.rst

You can then check if the email is present in the web interface at http://127.0.0.1:8025

On Bare Metal
=============

> If you do want to install the parsec server through docker, you can install it bare metal.

Reviewer: ```suggestion
If you don't want to install Parsec server via Docker, you can install it on bare metal.
```

Requirements
------------

- Python v3.9 with ``pip`` and ``venv`` modules
- A S3 like object storage endpoint.
- A postgres-14 database endpoint.

> - A TLS certificate & key for the server.

Reviewer: @Landeers what do you think? We can use the recommendations from the kickoff meeting.

> - A TLS certificate & key for the server.

Author: I understand why you would like to have the hardware requirement listed but consider:

1. The hardware requirement like depends on the usage of the server.
2. We don't have on our side a SIEM that would trace the usage of the server per user connected or something like that (I think of a Grafana graph). 

   With a SIEM that graph those data, we could obtain something like:

   - In term of CPU, we need ${cpu\\_per\\_user} * {users}$
   - In term of memory, we need ${ram\\_per\\_user} * {users}$

   Those value could also be obtain if we do a benchmark of the server :thinking:, but with the SIEM we would have a more realistic server load.

> - A TLS certificate & key for the server.

Reviewer: We do have estimates for cpu, ram and metadata/data storage based on our SaaS server. But you're right that it depends on type of usage. 

Maybe we can add that information later, after we have defined some standard use cases and give recommendations for those use cases.

Configure the environment
-------------------------

1. Configure the env files, follow `the env files`_.

Installation
------------

1. Set up a virtual env:

  .. code-block:: bash

    python -m venv venv

2. Configure your shell to use the virtual env:

  .. code-block:: bash

    source venv/bin/activate

3. Install ``parsec-server``

  The ``parsec-server`` is available as a python package hosted on https://pypi.org/project/parsec-cloud/.

  You need to install it with the extra ``backend`` enabled.

  .. code-block:: bash

    pip install 'parsec-cloud[backend]==2.16.0-a.0+dev'

4. Prepare the database by applying the migrations:

  .. code-block:: bash

    source venv/bin/activate
    set -a
    source parsec-db.env

>     python -m parsec.cli backend migrate

Author: Yes

Start the server
----------------

> 1. Create a wrapper script ``parsec-server-wrapper``

Author: done

  .. code-block:: bash

    #!/bin/bash

    # Load the virtualenv.
    source venv/bin/activate

    # Load the env file into the environment table.
    set -a
    for env_file in parsec-admin-token.env parsec-db.env parsec-smtp.env parsec-s3.env parsec.env; do
      source $env_file

>     done

Author: done
